src/utils/compiler/starknet_compile.py

```diff
@@ -0,0 +1,57 @@
+import json
+import os
+from dataclasses import dataclass
+from typing import List, Tuple
+
+from starkware.cairo.lang.cairo_constants import DEFAULT_PRIME
+from starkware.cairo.lang.compiler.cairo_compile import (
+    get_module_reader,
+)
+from starkware.cairo.lang.compiler.constants import MAIN_SCOPE, LIBS_DIR_ENVVAR
+from starkware.cairo.lang.compiler.preprocessor.preprocess_codes import preprocess_codes
+from starkware.starknet.compiler.compile import assemble_starknet_contract
+from starkware.starknet.compiler.starknet_pass_manager import starknet_pass_manager
+
+
+@dataclass
+class CairoSourceFile:
```

Comment:
Users would need to import it to use it with our SDK, let's stick to regular data structures. I guess we can replace it with `TypedDict`.